emt4j-analysis/src/main/java/org/eclipse/emt4j/analysis/autofix/FullAutofixExecutor.java

```diff
@@ -0,0 +1,129 @@
+import org.openrewrite.java.migrate.CastArraysAsListToList;
+import org.openrewrite.java.migrate.UseJavaUtilBase64;
+
+import java.util.*;
+
+
+// This executor contains most recipes that we support
+public class FullAutofixExecutor extends BaseAutofixExecutor {
+    public static final String[] JAVAEE_DEPENDENCIES = new String[]{
+            "com.sun.activation:javax.activation:jar:1.2.0",
+            "javax.transaction:javax.transaction-api:jar:1.2",
+            "javax.xml.bind:jaxb-api:jar:2.3.0",
+            "javax.xml.soap:javax.xml.soap-api:jar:1.4.0",
+            "com.sun.xml.bind:jaxb-core:jar:2.3.0",
+            "com.sun.xml.bind:jaxb-impl:jar:2.3.0",
+            "com.sun.xml.ws:jaxws-ri:pom:2.3.3",
+            "javax.annotation:javax.annotation-api:jar:1.3.2",
+    };
+
+    private static final DependencyUpdateRule[] FAMOUS_DEPENDENCY_UPDATE_RULES = {
+            new GATChangeDependencyUpdateRule("net.sourceforge.pinyin4j:pinyin4j:jar", "io.github.tokenjan:pinyin4j:jar:2.6.1"),
+            new GADontChangeDependencyUpdateRule("org.projectlombok:lombok:jar:1.18.22"),
+            new GADontChangeDependencyUpdateRule("org.apache.commons:commons-lang3:jar:3.12.0"),
+            new GADontChangeDependencyUpdateRule("org.asynchttpclient:async-http-client:jar:2.12.3"),
+            new GATChangeDependencyUpdateRule("org.mapstruct:mapstruct-jdk8:jar", "org.mapstruct:mapstruct:jar:1.5.5.Final"),
+            new GADontChangeDependencyUpdateRule("org.mapstruct:mapstruct-processor:jar:1.5.5.Final"),
+            new GADontChangeDependencyUpdateRule("commons-lang:commons-lang:jar:2.6"),
+
+    };
+
+    public FullAutofixExecutor(AutofixConfig config, MavenSession session, MavenProject project, Log log, RuntimeInformation runtime, SettingsDecrypter settingsDecrypter) {
+        super(config, session, project, log, runtime, settingsDecrypter);
+    }
+
+    @Override
+    protected boolean needPrescanSourceFiles() {
+        return true;
+    }
+
+    private PomUpdatePlan generatePomUpdatePlan() {
+        PomUpdatePlanGenerator pomUpdateGenerator = new PomUpdatePlanGenerator();
+
+        // update dependencies that we know they should be updated
+        for (DependencyUpdateRule rule : FAMOUS_DEPENDENCY_UPDATE_RULES) {
+            pomUpdateGenerator.updateDependency(rule, true);
+        }
+
+        // javaee packages
+        for (String dependency : JAVAEE_DEPENDENCIES) {
+            pomUpdateGenerator.addDependency(dependency);
+        }
+        // we need this to decide if a check can be autofixed
+        setFromGatv2toGatv(pomUpdateGenerator.getFromGatv2toGatv());
+
+        return pomUpdateGenerator.generatePlan();
+    }
+
+    @Override
+    protected Recipe getRecipe(Environment env) {
+        AutofixReport report = AutofixReport.getInstance();
+        List<String> recipesNames = new ArrayList<>();
+        List<Recipe> recipes = new ArrayList<>();
+
+        if (config.getToVersion() == 11) {
```

Comment:
We should select recipes based on both from and to version.

Reply:
now we select recipes based on both from and to version.